flocker/node/agents/blockdevice.py

```diff
@@ -1408,7 +1430,8 @@ def from_state_and_config(discovered_dataset, desired_dataset):
         Discovered.ATTACHED_ELSEWHERE: ActionNeeded,
         Discovered.ATTACHED_NO_FILESYSTEM: CreateFilesystem,
         Discovered.NON_MANIFEST: AttachVolume,
-        DatasetStates.ATTACHED: MountBlockDevice,
+        Discovered.ATTACHED: MountBlockDevice,
+        Discovered.ATTACHED_TO_DEAD_NODE: DetachVolume,
```

Comment:
No action required: This looks too easy, can we go back to a bunch of convoluted code across multiple functions for this sort of change? :stuck_out_tongue: Thanks @tomprince for the cleanup work!